Bug 1943050 - new `userScripts` API content (#38073)
mdn/content: files/en-us/mozilla/add-ons/webextensions/api/userscripts/resetworldconfiguration/index.md

---
title: userScripts.resetWorldConfiguration()
slug: Mozilla/Add-ons/WebExtensions/API/userScripts/resetWorldConfiguration
page-type: webextension-api-function
browser-compat: webextensions.api.userScripts.resetWorldConfiguration
---

{{AddonSidebar}}

Resets the configuration of a `USER_SCRIPT` world set by {{WebExtAPIRef("userScripts.configureWorld")}} to the defaults specified in {{WebExtAPIRef("userScripts.WorldProperties")}}. When the default world is reset, all worlds without an explicit configuration are also reset.

Any scripts injected into a reset world are moved to the default `USER_SCRIPT` world.

This is an asynchronous method that returns a {{JSxRef("Promise")}}.

## Syntax

```js-nolint

> const resettingWorldConfiguration = await browser.userScripts.resetWorldConfiguration(

Author: Done

---
title: userScripts.resetWorldConfiguration()
slug: Mozilla/Add-ons/WebExtensions/API/userScripts/resetWorldConfiguration
page-type: webextension-api-function
browser-compat: webextensions.api.userScripts.resetWorldConfiguration
---

{{AddonSidebar}}

Resets the configuration of a `USER_SCRIPT` world set by {{WebExtAPIRef("userScripts.configureWorld")}} to the defaults specified in {{WebExtAPIRef("userScripts.WorldProperties")}}. When the default world is reset, all worlds without an explicit configuration are also reset.

> Changes to world configurations only apply to new instances of the world: A configuration won't apply to a world initialized by the execution of a user script in a document until the document is reloaded. However, the browser may revoke certain privileges when a configuration is updated. For example, message calls from a `USER_SCRIPT` world may fail when `messaging` is reset to `false`.

Author: And here
```suggestion
Changes to world configurations only apply to new instances of the world: A configuration won't apply to a world initialized by the execution of a user script in a document until the document is reloaded. However, if the extension sets `messaging` to `false`, message calls from an update `USER_SCRIPT` world fail.
```

Reviewer: Already responded at https://github.com/mdn/content/pull/38073#discussion_r1966520227, in short: "may" is the correct qualifier.
